lib/logstash/inputs/jdbc.rb

```diff
@@ -220,6 +250,10 @@ def execute_query(queue)
     # update default parameters
     @parameters['sql_last_value'] = @sql_last_value
     execute_statement(@statement, @parameters) do |row|
+      if enable_encoding?
```

Comment:
jruby should be smart enough to inline this eventually. anyway I see your point but I'll leave as is for now.